pkg/reconciler/reconciler.go

```diff
@@ -20,51 +21,71 @@ type Reconciler interface {
 // Updater simply updates a resource by setting a controller reference
 // for resourceOwner and calling Patch on it.
 type Updater struct {
+	logger        logr.Logger
 	resourceOwner metav1.Object
 	resource      client.Object
 }
 
 func (r Updater) Reconcile(ctx context.Context, c client.Client, scheme *runtime.Scheme) error {
+
 	if r.resourceOwner.GetNamespace() == r.resource.GetNamespace() {
 		if err := controllerutil.SetControllerReference(r.resourceOwner, r.resource, scheme); err != nil {
+			r.logger.Info("set controller reference failed", "error", err)
```

Comment:
Why log the error and return it to the caller? In general I'm a big proponent of "Only handle errors once" (https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully).

Reply:
The issue is that when error gets logged, we have no idea `which` reconciler had issues reconciling `what`.

e.g. see if you can find the issue here and reason out why: https://github.com/rhobs/observability-operator/actions/runs/3231587984/jobs/5291269766

Reply:
Case in point, ... 
https://github.com/rhobs/observability-operator/actions/runs/3232742256/jobs/5293721088#step:4:971

```
2022-10-12T07:23:49Z	INFO	thanos-querier	watched MonitoringStack changed, checking for matching querier	{"Monitoring Stack": "e2e-tests/valid-loglevel"}
2022-10-12T07:23:49Z	INFO	observability-operator	patch failed	{"stack": "e2e-tests/valid-loglevel", "reconciler": "updater", "name": "valid-loglevel-alertmanager", "ns": "e2e-tests", "error": "roles.rbac.authorization.k8s.io \"valid-loglevel-alertmanager\" not found"}
2022-10-12T07:23:49Z	INFO	observability-operator	Failed to get prometheus object	{"stack": "e2e-tests/valid-loglevel", "err": "Prometheus.monitoring.rhobs \"valid-loglevel\" not found"}
2022-10-12T07:23:49Z	ERROR	Reconciler error	{"controller": "monitoringstack", "controllerGroup": "monitoring.rhobs", "controllerKind": "MonitoringStack", "monitoringStack": {"name":"valid-loglevel","namespace":"e2e-tests"}, "namespace": "e2e-tests", "name": "valid-loglevel", "reconcileID": "a60ef5f6-b16e-4dc4-82b5-91ad97f4f3f4", "error": "roles.rbac.authorization.k8s.io \"valid-loglevel-alertmanager\" not found"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:273
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:234
2022-10-12T07:23:49Z	INFO	observability-operator	Reconciling monitoring stack	{"stack": "e2e-tests/valid-loglevel"}
```

* The CI run failed with the following error - NOTE: `reconciler` is `updater`  and place it failed was when calling `Patch` which is a surprise (to me).

Reply:
My idea is that the error message should be as descriptive as possible. For this particular example:

```
return fmt.Errorf("%s/%s (%s): failed to set owner reference: %w", r.resource.GetNamespace(), r.resource.GetName(), r.resource.GetObjectKind().GroupVersionKind().String(), err)
```

Reply:
Yes, that will work and I will revert adding logger to including these details in the error msg. Thanks Simon 🤗